src/app/commands/cmd_remove_layer.cpp

```diff
@@ -11,20 +11,89 @@
 
 #include "app/app.h"
 #include "app/commands/command.h"
+#include "app/cmd/remove_tileset.h"
 #include "app/context_access.h"
 #include "app/doc_api.h"
 #include "app/i18n/strings.h"
 #include "app/modules/gui.h"
 #include "app/tx.h"
 #include "app/ui/status_bar.h"
 #include "doc/layer.h"
+#include "doc/layer_tilemap.h"
 #include "doc/sprite.h"
+#include "doc/tilesets.h"
 #include "fmt/format.h"
 #include "ui/alert.h"
 #include "ui/widget.h"
 
 namespace app {
 
+// If the UI is available this function shows an alert informing the user that they
+// cannot remove all the layers from the sprite (this is true when topLevelLayersToDelete
+// value is equals to the number of layers in the sprite's root group) and returns true.
+// If the UI is not available, this function just returns true if topLevelLayersToDelete is
+// equals to the number of layers in the sprite's root group.
+static bool deletingAllLayers(Context* ctx, Sprite* sprite, int topLevelLayersToDelete)
+{
+#ifdef ENABLE_UI
+  if (ctx->isUIAvailable() && topLevelLayersToDelete == sprite->root()->layersCount()) {
+    ui::Alert::show(Strings::alerts_cannot_delete_all_layers());
+  }
+#endif
+
+  return topLevelLayersToDelete == sprite->root()->layersCount();
+}
```

Comment:
Minor change
```suggestion
{
  const bool cancelDelete = (topLevelLayersToDelete == sprite->root()->layersCount());

#ifdef ENABLE_UI
  if (ctx->isUIAvailable() && cancelDelete) {
    ui::Alert::show(Strings::alerts_cannot_delete_all_layers());
  }
#endif

  return cancelDelete;
}
```

Reply:
I agree about the change, but I would like to name the bool "deletingAll" instead of "cancelDelete" if it makes sense to you

Reply:
Yes, it can be `deletingAll` 👍  just one thing I forgot: we prefer `static_functions_with_this_naming_convention` to avoid confusing them with member functions (methods) and var names.